[RSXIX] iP

RSXIX opened the pull request:
# Duke Pro

> "Your mind is for having ideas, not holding them." – David Allen | [source](https://dansilvestre.com/productivity-quotes)

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1.download it
1.double-click it.
1.add your tasks
1.let it manage your tasks for you :zany_face:

And it is **FREE**!

Features:
 - [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

RSXIX committed:
Completed Level-0

RSXIX committed:
Completed Level-1

RSXIX committed:
Completed Level-2

RSXIX committed:
Completed Level-3

RSXIX committed:
Completed Level-4

RSXIX committed:
Completed Level-5

RSXIX committed:
Completed Level-6

RSXIX committed:
Logically seems fine to me but not saving yet....

RSXIX committed:
Finished Level-7

RSXIX committed:
Merged with Level-7 Branch having Completed Level-7

RSXIX committed:
Revert "Merged with Level-7 Branch having Completed Level-7"

This reverts commit f0982afb694c6c27fbc24b7c50d20b0fcdaa3e29.

RSXIX committed:
Revert "Revert "Merged with Level-7 Branch having Completed Level-7""

This reverts commit 7e1e7e241228c3cbacb05da9fd65abe6a7821f3a.

Made a mistake

RSXIX committed:
Before Merge

RSXIX committed:
Merge branch 'master' into branch-Level-7

Seemed to be an issue

RSXIX committed:
Completed Level-8

RSXIX committed:
Completed A-MoreOOP

RSXIX committed:
Completed A-Packages

yucongkoo commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,20 @@
+
+import java.util.Scanner;
+import Duke.Tasks.*;
+import Duke.Exceptions.*;
```
I think that specifying the modules imported here might be more concise.

RSXIX replied:
Yeah I have removed wildcard imports

yucongkoo commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,20 @@
+        Ui iu = new Ui();
+        Parser her = new Parser();
+        TaskList tasks = new TaskList();
+        boolean running = true;
```
Will using "isRunning" rather than "running" make the name sound more boolean here?

yucongkoo commented on src/main/java/Duke/Storage.java:
```diff
@@ -0,0 +1,84 @@
+package Duke;
+
+import java.io.*;
```
...I noticed the same issue in other places too.

yucongkoo commented on src/main/java/Duke/Storage.java:
```diff
@@ -0,0 +1,84 @@
+import java.util.Arrays;
+import java.util.List;
+import Duke.Tasks.*;
+import Duke.Exceptions.*;
```
...I noticed the same issue in other places too.

yucongkoo commented on src/main/java/Duke/Tasks/DeadlineTask.java:
```diff
@@ -0,0 +1,26 @@
+    public String toString() {
+        return String.format("[D] | %s | %s | %s",
+                this.isDone() ? "[X]" : "[ ]",
+                str, Parser.dateToString(deadline));
```
Will it be better if the comma follows the next line in a line break. i.e. ```, this.isDone() ? "[X]" : "[ ]"``` for the second line?

yucongkoo commented on src/main/java/Duke/Tasks/EventTask.java:
```diff
@@ -0,0 +1,24 @@
+    public String toString() {
+        return String.format("[E] | %s | %s | %s - %s",
+                this.isDone() ? "[X]" : "[ ]",
+                event, Parser.dateToString(from.trim()), Parser.dateToString(to.trim()));
```
Will it be better if the comma follows the next line in a line break. i.e. ```, this.isDone() ? "[X]" : "[ ]"``` for the second line?

yucongkoo commented on src/main/java/Duke/Tasks/Task.java:
```diff
@@ -0,0 +1,44 @@
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+public class Task {
+    private Boolean done = false;
```
Will using isDone rather than done make the name some more boolean.

RSXIX replied:
Noted and I tried to make some of the variable names more intuitive or added comments to clarify

yucongkoo reviewed (commented):
Overall most of the code did follow the java coding standards, just some minor changes required.

RSXIX committed:
Merge branch 'add-gradle-support' into A-Gradle

RSXIX committed:
Added JUnit File Structure

RSXIX committed:
A-Gradle

RSXIX committed:
Revert "A-Gradle"

This reverts commit 03a4240878969bc7ba62769c4c6ebce7efccbebd.

RSXIX committed:
Completed A-JUnit

RSXIX committed:
Completed A-Jar

RSXIX committed:
Completed A-Find

RSXIX committed:
Did Java Docs for the inner packages

RSXIX committed:
Finished A-JavaDocs

RSXIX committed:
Completed A-CodingStandard

RSXIX committed:
Merge branch 'A-JavaDocs'

RSXIX committed:
Finished Merging Parallel Branches

Ruizhi2001 commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,125 @@
+     * @param iu The Ui object to interact with the user.
+     * @return True if the application should continue running; false if it should exit.
+     */
+    public boolean inputs(String input, TaskList tasks, Ui iu) {
```
Will it be better if the method name is a verb?

Ruizhi2001 commented on src/main/java/Duke/Storage.java:
```diff
@@ -0,0 +1,133 @@
+                if (splitsy.size() < 4) {
+                    splitsy.add("hi");
+                }
+                tasks.add(Storage.correctTask(splitsy.get(0), splitsy.get(1), splitsy.get(2), splitsy.get(3)));
```
Could this complicated expression be simplified and made more readable if you declared what each "splitsy.get(*)" means?

Ruizhi2001 commented on src/main/java/Duke/Tasks/TaskList.java:
```diff
@@ -0,0 +1,114 @@
+        String firstWord = item.split(" ")[0];
+
+        if (item.split(" ").length == 1) {
+            if (firstWord.equals("todo") || firstWord.equals("deadline") || firstWord.equals("event")) {
```
Could this expression be made simpler and more readable?

Ruizhi2001 commented on src/test/java/Duke/ParserTest.java:
```diff
@@ -0,0 +1,31 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+public class ParserTest {
+
+    public void testIncorrectInput() {
```
I think you might have forgotten to label with "@Test"

Ruizhi2001 reviewed (commented):
Just some minor updates. LGTM

RSXIX committed:
Implemented Feedback from Peer Review

RSXIX committed:
Completed Level-10

RSXIX committed:
Commiting so I can merge GUI into Master

RSXIX committed:
Merge branch 'branch-Level-10'

RSXIX committed:
Create new JAR file with Duke.Launcher entry point

This commit adds a new JAR file to the project, with Duke.Launcher specified as the entry point.

This follows the change from a text bot where Duke.Duke was the entry point to now a GUI.

The JAR file is created to streamline the execution of the Duke application.

[Include any additional details or remarks here if necessary]

Signed-off-by: [Your Name] <youremail@example.com>

RSXIX committed:
Improve code quality: General code enhancements

This commit aims to enhance code quality by critically examining the codebase
and making necessary improvements. The changes made in this commit may include
but are not limited to readability improvements in naming
and adherence to coding standards.

The goal is to ensure that the codebase maintains high quality and readability
to facilitate easier maintenance and collaboration among developers.

RSXIX committed:
Added More Code Enhancements

RSXIX committed:
Use Assertions to Document Critical Assumptions

Implement assert statements to document and validate critical assumptions
that should hold true at various points in the code. This practice enhances
code reliability and clarity during development and testing phases.

The added assertions help catch logical errors early and serve as valuable
documentation for future developers, highlighting essential assumptions.

RSXIX committed:
Merge pull request #1 from RSXIX/A-Assertions into Master

Implements Assertions to Document Critical Assumptions

RSXIX committed:
Merged Using Pull Requests A-Assertions into Master

RSXIX committed:
Merge pull request #2 from RSXIX/A-CodeQuality

A-CodeQuality

RSXIX committed:
Merged A-CodeQuality into Master

RSXIX committed:
Completed C-DetectDuplicates

RSXIX committed:
Completed A-UserGuide

RSXIX committed:
Added Screenshot Ui.png to docs folder

RSXIX committed:
Changed Duplicate Handling to a Hash Set, Added New Tests (which pass but still troubleshooting environment) and Also Added More Assert Test

RSXIX committed:
Fixing Environment Issue

RSXIX committed:
Completed A-Release
